### PR TITLE
feature: add network battle interface test

### PR DESCRIPTION
**Pull request by YuriCat**
environment test with network battle interface

### CODE REVIEW

File: tests/test_environment.py [hunk 1]
```diff
@@ -35,3 +38,37 @@ def test_environment(environment_path, env):
         traceback.print_exc()
 
     assert no_error_loop
+
+
+@pytest.mark.parametrize('env', ENVS)
+def test_environment2(environment_path, env):
```
Comment (ikki407) [inline tests/test_environment.py:44]: How about
```
def test_environment_network(environment_path, env):
    """Test battle loop using network battle interface of environment"""
```

Then, the old one should be the 'local' version?
```
def test_environment_local(environment_path, env):
    """Test battle loop using local battle interface of environment"""
```

File: tests/test_environment.py [hunk 2]
```diff
@@ -35,3 +38,37 @@ def test_environment(environment_path, env):
...
+                for player in e.turns():
+                    action = random.choice(es[player].legal_actions(player))
+                    actions[player] = es[player].action2str(action, player)
+                actions = {p: e.str2action(a, p) for p, a in actions.items()}
+                e.steps(actions)
+                for p, e_ in es.items():
+                    info = e.diff_info(p)
+                    e_.update(info, False)
+                e.reward()
+            e.outcome()
```
Comment (ikki407) [inline tests/test_environment.py:68]: I think this is a bit weird... What do you think?

At least, should the returned value be set something like `_`? (Although this test aims to test not to occur any errors... so the current test could be okay)
```
_ = e.outcome()
```

File: tests/test_environment.py (hunk 1 above)
Comment (YuriCat) [reply tests/test_environment.py:45]: Changing names is good idea.

File: tests/test_environment.py (hunk 2 above)
Comment (YuriCat) [reply tests/test_environment.py:69]: I don't think this current code is strange, but it's ok for me to change these lines as such.

Comment (ikki407) [reply tests/test_environment.py:69]: Ok, just leave it as it is

**Comment by ikki407** [comment]
LGTM

**Pull request merged**